tibanna/cw_utils.py

```diff
@@ -324,19 +324,117 @@ def max_ebs_read_used_all_pts(self):
         pts = [(r['Average'], r['Timestamp']) for r in res['Datapoints']]
         return[p[0] for p in sorted(pts, key=lambda x: x[1])]
 
+    
+    @staticmethod
+    def extract_metrics_data(file_contents):
+      # This function takes a file contents string and parses it column wise.
+      # It returns the header and the data in a way that can be injected into JS code
+      # `data` is a Python dict, that contains all that data for later processing
+      # outside of this function.
+      columns = []
+      columns_js = "[]"
+      data = {}
+      data_js = "[]"
+
+      is_header_line = True
+      for line in file_contents.rstrip().split('\n'):
+          if is_header_line:
+              for k,col in enumerate(line.split('\t')):
+                  columns.append(col)
+                  data[col] = []
+              is_header_line = False
```

Comment:
I would avoid a boolean trigger pattern like this and instead enumerate on the lines (you know index=0 is the header):
```
for idx, line in enumerate(file_contents.rstrip().split('\n')):
    if idx == 0:  # is_header_line
        for k,col in enumerate(line.split('\t')
            columns.append(col)
            data[col] = []
    else: ...
```